net-libs/mbedtls/mbedtls-2.6.0.recipe

```diff
@@ -0,0 +1,139 @@
+		cmd:mbedtls_gen_entropy = $portVersion
+		cmd:mbedtls_gen_key = $portVersion
+		cmd:mbedtls_gen_random_ctr_drbg = $portVersion
+		cmd:mbedtls_gen_random_havege = $portVersion
+		cmd:mbedtls_generic_sum = $portVersion
+		cmd:mbedtls_hello = $portVersion
+		cmd:mbedtls_key_app = $portVersion
+		cmd:mbedtls_key_app_writer = $portVersion
+		cmd:mbedtls_mini_client = $portVersion
+		cmd:mbedtls_mpi_demo = $portVersion
+		cmd:mbedtls_pem2der = $portVersion
+		cmd:mbedtls_pk_decrypt = $portVersion
+		cmd:mbedtls_pk_encrypt = $portVersion
+		cmd:mbedtls_pk_sign = $portVersion
+		cmd:mbedtls_pk_verify = $portVersion
+		cmd:mbedtls_req_app = $portVersion
+		cmd:mbedtls_rsa_decrypt = $portVersion
+		cmd:mbedtls_rsa_encrypt = $portVersion
+		cmd:mbedtls_rsa_genkey = $portVersion
+		cmd:mbedtls_rsa_sign = $portVersion
+		cmd:mbedtls_rsa_sign_pss = $portVersion
+		cmd:mbedtls_rsa_verify = $portVersion
+		cmd:mbedtls_rsa_verify_pss = $portVersion
+		cmd:mbedtls_selftest = $portVersion
+		cmd:mbedtls_ssl_cert_test = $portVersion
+		cmd:mbedtls_ssl_client1 = $portVersion
+		cmd:mbedtls_ssl_client2 = $portVersion
+		cmd:mbedtls_ssl_fork_server = $portVersion
+		cmd:mbedtls_ssl_mail_client = $portVersion
+		cmd:mbedtls_ssl_server = $portVersion
+		cmd:mbedtls_ssl_server2 = $portVersion
+		cmd:mbedtls_strerror = $portVersion
+		cmd:mbedtls_udp_proxy = $portVersion
+		"
+fi
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	mbedtls${secondaryArchSuffix}_devel = $portVersion
+	devel:libmbedcrypto$secondaryArchSuffix = 0 compat >= 0
+	devel:libmbedtls$secondaryArchSuffix = 10 compat >= 10
+	devel:libmbedx509$secondaryArchSuffix = 0 compat >= 0
+	"
+REQUIRES_devel="
+	mbedtls$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:perl
+	cmd:sed
+	"
+
+PATCH()
+{
+	sed -i \
+		-e "s|/include/mbedtls|/$relativeOldIncludeDir|;" \
+		-e "s|/include|/$relativeOldIncludeDir|;" \
+		-e "s|/bin/|/$relativeBinDir/|;" \
+		-e "s|/bin\$|/$relativeBinDir/|;" \
+		-e "s|/lib\$|/$relativeLibDir/|;" \
+		Makefile
+}
+
+if [ -n "$secondaryArchSuffix" ]; then
+	maybe_binDir_mbedtls_runtimes=
+else
+	maybe_binDir_mbedtls_runtimes="`echo "$PROVIDES" | sed -n \
+		-e "s|cmd:\([^ ]*\).*|$binDir/\1|p" \
+		`"
+fi
+defineDebugInfoPackage mbedtls$secondaryArchSuffix \
+	$libDir/libmbedcrypto.so.0 \
+	$libDir/libmbedtls.so.10 \
+	$libDir/libmbedx509.so.0 \
+	$maybe_binDir_mbedtls_runtimes
+
+BUILD()
+{
+	# Haiku is unix-like and network functions are in libnetwork
+	make CFLAGS="-Dunix" LDFLAGS="-lnetwork" SHARED=yes WARNING_CFLAGS=""
+}
+
+INSTALL()
+{
+	make install SHARED=yes DESTDIR=$prefix
+	prepareInstalledDevelLibs libmbedcrypto libmbedtls libmbedx509
+	packageEntries devel $developDir
+	if [ -n "$secondaryArchSuffix" ]; then
+		rm -rf $binDir
+	fi
+}
+
+TEST()
+{
+	LIBRARY_PATH="$sourceDir/library:$LIBRARY_PATH" make check
```

Comment:
Thanks for you update!

If you have ed installed, could you please run
```
(echo 96,106 m 119; echo w; echo q) | ed mbedtls-2.6.0.recipe
```
It will fix a small error, due to a wrong indication I gave you in my previous review.
The result is that the PATCH fonction will be moved after the call to defineDebugInfoPackage.
It's my fault, sorry.

While at it, would you mind doing an extra cosmetic change? Could you please replace, in `TEST()`
```
	LIBRARY_PATH="$sourceDir/library:$LIBRARY_PATH" make check
```
by
```
	LIBRARY_PATH="$sourceDir/library${LIBRARY_PATH:+:$LIBRARY_PATH}" make check
```
Yep, this is really cosmetics. But it looks nicer ;-)
It does not make any difference when LIBRARY_PATH is not an empty string.
But if LIBRARY_PATH is empty, then it makes a tiny difference, since, during the call to "make check" LIBRARY_PATH won't end with a colon. That's cosmetics, but it will be cool ;-)
Thanks!

Reply:
I find that this makes the line less readable, so I'd rather avoid it.

Reply:
I don't really care.
It was just to make sure no purist would be able to say "this is not 100% correct" ;-)